lib/charms/tls_certificates_interface/v0/tls_certificates.py

```diff
@@ -449,6 +449,36 @@ def _parse_certificates_from_relation_data(relation_data: dict) -> List[Cert]:
                     )
         return certificates
 
+    def get_certificates_for_common_name(self, common_name: str = None, departed_id: int = None) -> List[Cert]:
+        """Loops over all relations and returns list of Cert objects.
+
+        Args:
+            common_name: return certificates for specified the common name.
+            departed_id: don't return certificates from departed relation.
+
+        Returns:
+            list: List of certificates.
+        """
+        certificates = []
+        relations = self.model.relations[self.relationship_name]
+        for relation in relations:
+            if departed_id is not None and relation.id == departed_id:
```

Comment:
```suggestion
            if relation.id == departed_id:
```

Reply:
fixed in https://github.com/canonical/tls-certificates-interface/pull/4/commits/79e1543206e3b7a6e79f06203a7ff4adc3c8c076